python/ql/lib/semmle/python/frameworks/Aiohttp.qll

```diff
@@ -664,6 +664,51 @@ module AiohttpWebModel {
     override DataFlow::Node getNameArg() { result in [this.getArg(0), this.getArgByName("name")] }
 
     override DataFlow::Node getValueArg() { result in [this.getArg(1), this.getArgByName("value")] }
+
+    override predicate hasSecureFlag(boolean b) {
+      super.hasSecureFlag(b)
+      or
+      exists(DataFlow::Node arg, BooleanLiteral bool | arg = this.getArgByName("secure") |
+        DataFlow::localFlow(DataFlow::exprNode(bool), arg) and
+        b = bool.booleanValue()
+      )
+      or
+      not exists(this.getArgByName("secure")) and
+      b = false
+    }
+
+    override predicate hasHttpOnlyFlag(boolean b) {
+      super.hasHttpOnlyFlag(b)
+      or
+      exists(DataFlow::Node arg, BooleanLiteral bool | arg = this.getArgByName("httponly") |
+        DataFlow::localFlow(DataFlow::exprNode(bool), arg) and
+        b = bool.booleanValue()
+      )
+      or
+      not exists(this.getArgByName("httponly")) and
+      b = false
+    }
+
+    override predicate hasSameSiteAttribute(Http::Server::CookieWrite::SameSiteValue v) {
+      super.hasSameSiteAttribute(v)
+      or
+      exists(DataFlow::Node arg, StringLiteral str | arg = this.getArgByName("samesite") |
+        DataFlow::localFlow(DataFlow::exprNode(str), arg) and
+        (
+          str.getText().toLowerCase() = "strict" and
+          v instanceof Http::Server::CookieWrite::SameSiteStrict
+          or
+          str.getText().toLowerCase() = "lax" and
+          v instanceof Http::Server::CookieWrite::SameSiteLax
+          or
+          str.getText().toLowerCase() = "none" and
+          v instanceof Http::Server::CookieWrite::SameSiteNone
+        )
+      )
+      or
+      not exists(this.getArgByName("samesite")) and
+      v instanceof Http::Server::CookieWrite::SameSiteLax // Lax is the default
+    }
```

Comment:
This code seems to be copied across a bunch of frameworks. Would it make sense to create a `SetCookieCall` abstraction so the frameworks just have to point out where the calls are? It seems all calls to `set_cookie` (in any framework) are likely to follow this convention (and if not, it can be overridden by the framework).

Reply:
Seems like it would make sense. Should it be defined in the `Concepts` library?

Reply:
Yes, since it comes from the protocol (something like [here]()https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie) it should probably not be in a specific framework, so if it could be in `Concepts`, that would be nice.